05/DB.test.js

```diff
@@ -7,43 +7,43 @@ describe('class DB', () => {
         return expect(db).toBeInstanceOf(DB);
     });
 
-    it('is database object', () => {
-        const db = new DB();
-        return expect(db).toBeInstanceOf(Object);
-    });
-
-    it('is array declared', () => {
-        const db = new DB();
-        return expect(db._rows).toBeTruthy();
-    });
-
-    it('is array object', () => {
-        const db = new DB();
-        return expect(db._rows).toBeInstanceOf(Object);
-    });
-
     it('is array empty', () => {
         const db = new DB();
         return expect(db._rows.length).toBe(0);
-      });
+    });
 
 });
 
 describe('data', () => {
 
-    it('if data.id is not exist', () => {
-        const db = new DB();
-        const data1 = {id: 2, a: 1, b: 1};
-        db.insert(data1);
+    it('rejects when id is duplicated', async () => {
+        const db = new DB;
+        await db.insert({id: 1, a: 'a', b: 'b'});
+        await db.insert({id: 2, a: 'a', b: 'b'});
+        await db.insert({id: 3, a: 'a', b: 'b'});
+        await db.insert({id: 4, a: 'a', b: 'b'});
+        await db.insert({id: 5, a: 'a', b: 'b'});
+        await db.insert({id: 6, a: 'a', b: 'b'});
+        await expect(db.insert({id: 5, a: 'a', b: 'b'})).rejects.toMatch('ID can\'t be duplicated!');
+    })
 
-        const promise = new Promise(function(resolve, reject) {
-            if (data1.id !== null) {
-                resolve(true);
-            } else {
-                reject('data.id is not exist!');
-            }
-        });
-    });
+    it('how many elements exist', async () => {
+        const db = new DB;
+        let numberOfElements = db._rows.length;
+
+        await db.insert({id: 1, a: 'a', b: 'b'});
+        await db.insert({id: 2, a: 'a', b: 'b'});
+        await db.insert({id: 3, a: 'a', b: 'b'});
+        await db.insert({id: 4, a: 'a', b: 'b'});
+        await db.insert({id: 5, a: 'a', b: 'b'});
+        await db.insert({id: 6, a: 'a', b: 'b'});
+
+        await db.remove(db._rows.length);
+        await db.remove(db._rows.length);
+
+        numberOfElements = db._rows.length;
+        await expect(db._rows.length).toBe(numberOfElements);
```

Comment:
tutaj zawsze będzie to samo bo to ten sam element (nawet jeśli zapisany do innej zmiennej), ale już `db.getRows()` byłoby uzasadnione ;)